Src/Scripts/Include-GenerateUnitClassSourceCode.ps1

```diff
@@ -278,11 +280,30 @@ namespace UnitsNet
         public static $className Parse(string str, IFormatProvider formatProvider = null)
         {
             if (str == null) throw new ArgumentNullException("str");
-            string[] words = str.Split(new[] {" "}, StringSplitOptions.RemoveEmptyEntries);
-            if (words.Length < 2)
+
+            var numFormat = formatProvider != null ?
+                (NumberFormatInfo) formatProvider.GetFormat(typeof (NumberFormatInfo)) :
+                (NumberFormatInfo) CultureInfo.CurrentCulture.NumberFormat.Clone();
+
+            var numRegex = @"[\d., "                        // allows digits, dots, commas, and spaces in the number by default
+                         + numFormat.NumberGroupSeparator   // adds provided (or current) culture's group separator
+                         + numFormat.NumberDecimalSeparator // adds provided (or current) culture's group separator
+                         + @"]*\d";                         // ensures quantity ends in digit
+            var regexString = @"\s*"                                                             // ignore leading whitespace
+                            + @"(?<value>[-+]?" + numRegex + @"(?:[eE][-+]?" + numRegex + @")?)" // capture Quantity input
+                            + @"\s?"                                                             // ignore whitespace (if any)
```

Comment:
I was actually contemplating whether this should be /s? or /s*.
What would you suggest?